cpp/src/gandiva/llvm_generator.cc

```diff
@@ -369,9 +476,43 @@ void LLVMGenerator::ComputeBitMapsForExpr(const CompiledExpr& compiled_expr,
   // Extract the destination bitmap address.
   int out_idx = compiled_expr.output()->validity_idx();
   uint8_t* dst_bitmap = eval_batch.GetBuffer(out_idx);
-
+  int64_t num_records = eval_batch.num_records();
+  LocalBitMapsHolder bit_map_holder = LocalBitMapsHolder(num_records, 1);
+  uint8_t* temp_bitmap = bit_map_holder.GetLocalBitMap(0);
   // Compute the destination bitmap.
-  accumulator.ComputeResult(dst_bitmap);
+  switch (mode) {
+    case SelectionVectorMode::NONE: {
+      accumulator.ComputeResult(dst_bitmap);
+      return;
+    }
+    case SelectionVectorMode::INT16: {
+      accumulator.ComputeResult(temp_bitmap);
```

Comment:
the three case statements below are exactly same. you could modularize using a template function (or selection vector).

Reply:
another alternative is to copy only the set bits from the temp_bitmap to dst_bitmap.